### ISSUE TITLE
Wildcard query fails for undefined elements

### ISSUE THREAD

**Issue by catobus** [mmckegg/json-query#22]
Hi,

I am using version 2.1.0. This is a simplified version of JSON that I am using: 

```
var image = {
   "sources" : [
      {
         "name": "source1",
         "items": [
            {"name": "source1.item1"},
         ]
      }]
};

```
_sources_ array is optional, and may be empty or not specified at all. My objective is to check whether there are any _source.items_ exist. So I am using this query: 

`jsonQuery('sources[*].items', {data: image});`

However, when sources array is not defined this query fails. I essentially execute the following:  

`jsonQuery('sources[*].items', {data: {}});`

and it fails with exception: 

```
C:\test\node_modules\json-query\index.js:72
    var keys = Object.keys(state.currentItem)
                      ^
TypeError: Cannot convert undefined or null to object
    at handleToken (C:\test\node_modules\json-query\index.js:72:23)
    at handleQuery (C:\test\node_modules\json-query\index.js:37:9)
    at jsonQuery (C:\test\node_modules\json-query\index.js:19:10)
    at Object.<anonymous> (C:\test\tester.js:64:14)
    at Module._compile (module.js:541:32)
    at Object.Module._extensions..js (module.js:550:10)
    at Module.load (module.js:456:32)
    at tryModuleLoad (module.js:415:12)
    at Function.Module._load (module.js:407:3)
    at Function.Module.runMain (module.js:575:10)
```

Is there any way to execute such query without previously checking that an array is defined? 

Thanks!

**Comment by mmckegg** [comment]
I think this is a bug. `json-query` should not be throwing errors if attributes are missing, just returning `null`.

**Comment by mmckegg** [comment]
This particular case is a little bit complicated though, because you are essentially doing a map. Should missing attributes be `null` or just excluded from the final result?

`[{"name": "source1.item1"}, null, {"name": "source2.item2"}]`

or

`[{"name": "source1.item1"}, {"name": "source2.item2"}]`

**Comment by catobus** [comment]
Logically, I would expect the missing attributes be excluded from the result. However, in this particular case I am processing the results further and nulls will be just ignored.

**Comment by mmckegg** [comment]
Yes, sorry now I see what's going on. This is much simpler than I thought 😄 

Fix is a few minutes away!

**Comment by catobus** [comment]
Thank you for a quick response and your help!

**Comment by mmckegg** [comment]
Published in **v2.1.1** 🎉
